include/async_scope.hpp

```diff
@@ -624,78 +289,69 @@ namespace _P2519::execution {
 
         using __op_t = __future_operation_t<_Sender>;
 
+        void __step_from_to_(std::unique_lock<std::mutex>& __guard, __future_state_steps __from, __future_state_steps __to) {
+          if (!__guard.owns_lock()) { std::terminate(); }
```

Comment:
My preference would be to turn these sanity checks into asserts. This lib should probably have a customizable assert macro (that defaults to `assert`?) so that users can control what happens on contract violations. Something like:

```c++
#ifdef _P2300_ASSERT
#error Hammer says, ya can't touch this
#endif

#define _P2300_ASSERT(_X) \
  do { \
    static_assert(noexcept(_X)); \
    STDEXEC_ASSERT(_X); \
  } while(false)

#ifndef STDEXEC_ASSERT
#define STDEXEC_ASSERT assert
#endif
```

It would cause an extra PP evaluation of the PP token sequence `_X`, but that's rarely an issue in practice.

This could live in the `__config.hpp` header.

Reply:
I did not find a `__config.hpp`, so I put them in `__utility.hpp` and `assert` has a bad default, so I defaulted to `std::terminate()`;